### PR TITLE
Refactor Project Structure to MVC

### PR DESCRIPTION
**Pull request by bmsant**
This pull request includes significant refactoring and improvements to the project structure:

- Refactoring to MVC: We have restructured the codebase using the Model-View-Controller (MVC) design pattern to improve maintainability and scalability.

Centralized Constants and Interfaces: Defined constants, instances, and interfaces to centralize elements that are repetitive or likely to change in the future.

**Changes**

Refactored the project structure to follow the MVC design pattern.
Centralized constants, instances, and interfaces for better maintainability.

**Notes**

Please review the changes and provide feedback.
Ensure that all tests pass before merging.
A subsequent pull request will be submitted where the new MVC structure was tested to ensure it functions correctly.

```
eokportal/
├── node_modules/
├── public/
│   ├── next.svg
│   ├── vercel.svg
│   └── ...
├── src/
│   ├── app/
│   │   ├── favicon.ico
│   │   ├── globals.css
│   │   ├── layout.tsx
│   │   └── page.tsx
│   ├── lib/
│   │   ├── actions/
│   │   │   ├── captcha.actions.ts
│   │   │   └── masterAccount.actions.ts
│   │   ├── models/
│   │   │   ├── captcha.models.ts
│   │   │   └── masterAccount.models.ts
│   │   ├── services/
│   │   │   └── captcha.ts
│   │   └── eokapi.ts
│   └── utils/
│       └── eokapi.ts
├── .gitignore
├── package.json
├── package-lock.json
├── postcss.config.mjs
├── tailwind.config.ts
└── tsconfig.json
```

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export const API_ENDPOINTS = {
+  CAPTCHA: "/api/v1/captchas",
+  MASTER_ACCOUNTS: {
+    BASE: "/api/v1/master-accounts", // POST: Create new account
```
Comment (naatsuo) [inline src/constants.js:4]: por que vc nao tipa poe exemplo

POST_CREATE
POST_LOGIN
POST_CHANGE_PASSWORD
...
PATCH_VERIFY_EMAIL_CODE
assim ate os outros

se tiver algum GET vc colocar o GET_...

Oq acha ?

File: src/lib/actions/captcha.actions.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import eokapi from "../../utils/eokapi";
+import { API_ENDPOINTS } from "@/constants";
+import { CaptchaResponse } from "../models/captcha.models";
+
+
+export async function getCaptcha(): Promise<CaptchaResponse> {
+  try {
+    const response = await eokapi.get<CaptchaResponse>(API_ENDPOINTS.CAPTCHA);
+    const data = response.data;
+    sessionStorage.setItem('token', data.token);
```
Comment (naatsuo) [inline src/lib/actions/captcha.actions.ts:10]: Geralmente eu gosto de soltar uma linha quando eu to criando uma variavel e depois chamo algum metodo, crio essa separação pra ficar mais legivel. 

e da um ar de menos poluido

File: src/lib/actions/masterAccount.actions.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+export async function loginMasterAccount(data: MasterAccountRequests) {
+  try {
+    const response = await eokapi.post(API_ENDPOINTS.MASTER_ACCOUNTS.LOGIN, {
+      email: data.email,
+      password: data.password,
+      captchaReply: data.captchaReply,
+    });
+    const responseData = response.data;
+    sessionStorage.setItem('token', responseData.token);
```
Comment (naatsuo) [inline src/lib/actions/masterAccount.actions.ts:28]: valida o pulo da linha

**Comment by naatsuo** [comment]
Alt Text

File: src/constants.js (hunk 1 above)
Comment (bmsant) [reply src/constants.js:4]: Depois eu faco isso.

**Review by bmsant — COMMENTED** [review]
Resolvendo os B.O

**Review by naatsuo — APPROVED** [review]

**Pull request merged**